mgmt: hawkbit: do not prepend custom controllerId

nealjack opened the pull request:
This PR changes how the controllerId is generated based on device id, and disentangles the two. The controllerId is what hawkbit uses to uniquely identify a device, and is not necessarily the same as the device id, and should be fully customizeable by the user if needed. Previously, all custom device ids were being prepended with `CONFIG_BOARD`. When a user selects `CONFIG_HAWKBIT_CUSTOM_DEVICE_ID`, they should be able to specify the full controllerId used with hawkbit, without a forced prepend.

The default handler (`hawkbit_get_device_identity_default`) for returning the controllerId is modified to also include the prefix, so the default behavior without `CONFIG_HAWKBIT_CUSTOM_DEVICE_ID` enabled is preserved.

nealjack committed:
mgmt: hawkbit: do not prepend custom controllerId

This commit changes how the controllerId is generated based on device
id, and disentangles the two. The controllerId is what hawkbit uses to
uniquely identify a device, and is not necessarily the same as the
device id, and should be fully customizeable by the user if needed.
Previously, all custom device ids were being prepended with
`CONFIG_BOARD`. When a user selects `CONFIG_HAWKBIT_CUSTOM_DEVICE_ID`,
they should be able to specify the full controllerId used with hawkbit,
without a forced prepend.

Signed-off-by: Neal Jackson <neal@blueirislabs.com>

maass-hamburg commented on subsys/mgmt/hawkbit/hawkbit_device.h:
```diff
@@ -18,6 +18,8 @@
 #define DEVICE_ID_HEX_MAX_SIZE	((DEVICE_ID_BIN_MAX_SIZE * 2) + 1)
 #endif
 
+#define CONTROLLER_ID_MAX_SIZE	(32 + DEVICE_ID_HEX_MAX_SIZE)
```
include it instead in the DEVICE_ID_HEX_MAX_SIZE for the non custom dev id and use sizeof(CONFIG_BOARD) instead of the arbitrary 32 and don't have a seperate CONTROLLER_ID_MAX_SIZE

maass-hamburg commented on subsys/mgmt/hawkbit/hawkbit.c:
```diff
@@ -1456,9 +1456,8 @@ static void s_report(void *o)
-	snprintk(url_buffer, sizeof(url_buffer), "%s/%s-%s/%s/%d/%s", HAWKBIT_JSON_URL,
-		 CONFIG_BOARD, s->device_id, "deploymentBase", s->hb_context.json_action_id,
-		 "feedback");
+	snprintk(url_buffer, sizeof(url_buffer), "%s/%s/%s/%d/%s", HAWKBIT_JSON_URL,
```
please add a entry in the migration guide, that the board is no longer prefixed automatically, when using a custom device id callback. The user has to add that to their callback if they want to preserver the old behavior.

maass-hamburg commented on subsys/mgmt/hawkbit/hawkbit.c:
```diff
@@ -143,7 +143,7 @@ struct hawkbit_context {
 	struct smf_ctx ctx;
 	struct hawkbit_context hb_context;
-	char device_id[DEVICE_ID_HEX_MAX_SIZE];
+	char controller_id[CONTROLLER_ID_MAX_SIZE];
```
in the Kconfigs it is still called device_id, maybe leave it.

rlubos commented:
About the compliance failure, note that the email address in the sign-off line should be inside square bracket, I. e. `Signed-off-by: Neal Jackson <neal@blueirislabs.com>` (git does this automatically if you use `git commit -s` parameter.

nealjack committed:
mgmt: hawkbit: add migration note for device_id prepend with custom id

When `CONFIG_HAWKBIT_CUSTOM_DEVICE_ID` is selected, the device_id will
no longer be automatically prepended with `CONFIG_BOARD`.

Signed-off-by: Neal Jackson <neal@blueirislabs.com>

maass-hamburg commented on subsys/mgmt/hawkbit/hawkbit_device.c:
```diff
@@ -21,15 +21,20 @@ static bool hawkbit_get_device_identity_default(char *id, int id_max_len)
+		return false;
+	}
+
+	length = bin2hex(hwinfo_id, (size_t)id_length, id + length, id_max_len - length);
```
```suggestion
	length = bin2hex(hwinfo_id, (size_t)id_length, &id[length], id_max_len - length);
```

maass-hamburg commented on subsys/mgmt/hawkbit/hawkbit.c:
```diff
@@ -1214,7 +1215,7 @@ static void s_start(void *o)
 	}
 
-	if (!hawkbit_get_device_identity(s->device_id, DEVICE_ID_HEX_MAX_SIZE)) {
+	if (!hawkbit_get_device_identity(s->device_id, DEVICE_ID_MAX_SIZE)) {
```
```suggestion
	if (!hawkbit_get_device_identity(s->device_id, DEVICE_ID_HEX_MAX_SIZE)) {
```

DEVICE_ID_MAX_SIZE is no longer defined

maass-hamburg commented on subsys/mgmt/hawkbit/hawkbit.c:
```diff
@@ -143,7 +143,7 @@ struct hawkbit_context {
 	struct smf_ctx ctx;
 	struct hawkbit_context hb_context;
-	char device_id[DEVICE_ID_HEX_MAX_SIZE];
+	char device_id[DEVICE_ID_MAX_SIZE];
```
```suggestion
	char device_id[DEVICE_ID_HEX_MAX_SIZE];
```

maass-hamburg commented:
@nealjack needs another rebase, due to some bug, that lets ci fail, that is fixed now on the current main

rlubos commented:
> @nealjack needs another rebase, due to some bug, that lets ci fail, that is fixed now on the current main

Apparently rerun was enough, no need to rebase.
